Second commit

popozarski opened the pull request.

Commit:
Second commit

Ayantea1 commented on src/test/java/MainPageTests.java:
```diff
@@ -0,0 +1,87 @@
+
+   }
+    @Test
+    public void checkTextFAQ() {
```
Нужно исправить: вопросы и ответы нужно проверить через параметризацию. По сути это отдельные тесты, если упадет один из них, остальные должны тоже быть проверены, а мы должны понимать, какой именно упал

popozarski replied:
Исправлено, добавил проверку на вопрос и ответ отдельно и параметризовал тест

Ayantea1 commented on src/test/java/MainPageTests.java:
```diff
@@ -0,0 +1,87 @@
+
+    // Проверка верхней кнопки "Заказать"
+    // Если кнопка работает, то меняется URL
+    @Test
```
Нужно исправить: тесты должны быть разнесены в разные классы по функционалу

popozarski replied:
Убрал тест на проверку кнопок "Заказать", кнопку добавил в качестве аргумента в параметры OrderTests, где кнопки заказа будут проверяться по ходу сценария

Ayantea1 commented on src/test/java/MainPageTests.java:
```diff
@@ -0,0 +1,87 @@
+    @Test
+    public void orderButtonTopRedirectToOrderForm() {
+        MainPage objMainPage = new MainPage(driver);
+        String expectedURL = "https://qa-scooter.praktikum-services.ru/order";
```
можно улучшить: урл лучше вынести в константу

popozarski replied:
Убрал совсем, нашел более оптимальное решение

Ayantea1 commented on src/test/java/MainPageTests.java:
```diff
@@ -0,0 +1,87 @@
+    // Проверка кнопки "Заказать" в середине страницы
+    // Если кнопка работает, то меняется URL
+    @Test
+    public void orderButtonMidRedirectToOrderForm() {
```
можно улучшить: тесты отличаются только кнопкой, их тоже можно параметризовать

Ayantea1 commented on src/test/java/pageObject/MainPage.java:
```diff
@@ -0,0 +1,88 @@
+
+    //-----------------------------------------------------------------------
+    //Кнопка заказать верхняя
+    private By orderButtonTop = By.xpath("//*[@id=\"root\"]/div/div/div[1]/div[2]/button[1]");
```
Нужно исправить: не должно быть локаторов, ищущих по порядку элементов. Он может измениться

popozarski replied:
Исправлено

Commit:
MainPage(FAQTests):
Тест MainPage переименован в FAQTests;
добавлен параметризованный тест для вопросов-ответов;
Тесты на кнопки "Заказать" перенесены в OrderTests (разнесены в разные классы по функционалу).
OrderTests:
В параметризованный тест добавлен аргумент кнопка "Заказать", чтобы сценарий соответствовал финальному заданию спринта
В сценарий теста PositiveFlow добавлен шаг нажатия на кнопки "Заказать".
pageObject/MainPage:
Исправлены локаторы с поиском по порядку элементов.

Ayantea1 commented on src/test/java/pageObject/OrderAcceptedWindow.java:
```diff
@@ -0,0 +1,31 @@
+    }
+    public String copyHeadlineText(){
+        //new WebDriverWait(driver,Duration.ofSeconds(3))
+        // .until(ExpectedConditions.visibilityOfElementLocated(acceptedOrderPopupWindow));
```
не забывай убирать неиспользуемый код
